pypeit/core/gui/gui_util.py

```diff
@@ -0,0 +1,257 @@
+
+        Args:
+            action (:obj:`str`):
+                The keyword for the event action to perform
+            func (callable):
+                The function to call when the corresponding event is triggered
+            descr (:obj:`str`, optional):
+                The description of the action being taken.  Used to build a help
+                dialog.
+        """
+        self.observers[action] = func
+        self.observers_descr[action] = descr
+
+    def disconnect(self, action):
+        """
+        Remove the action from the register.
+
+        Args:
+            action (:obj:`str`):
+                The keyword for the event action to remove
+        """
+        try:
+            del self.observers[action]
+        except KeyError:
+            pass
+        try:
+            del self.observers_descr[action]
+        except KeyError:
+            pass
+
+    def build_help(self):
+        """
+        Register the help dialog.  Any action already assigned to the '?' key
+        will be removed!
+        """
+        if '?' in self.observers:
+            msgs.warn('Help key (?) is already registered and will be overwritten')
+            self.disconnect('?')
+        self.register('?', self.print_help, descr='Print this list of key bindings')
+
+    def print_help(self, pos):
+        """
+        Print the help dialog.
+
+        Args:
+            pos (list):
+                Location where the help event was triggered.  This is ignored.
+        """
+        print('-'*50)
+        print('Key bindings')
+        for action in self.observers.keys():
+            descr = 'No description given' if self.observers_descr[action] is None \
+                        else self.observers_descr[action]
+            print(f'    {action}: {descr}')
+        print('-'*50)
+
+
+class UpdateableRangeSlider(widgets.RangeSlider):
+    """
+    A range slider.
+
+    This is virtually identical to the base class, but with a few customizations
+    regarding where labels are placed (or removed).
+    """
+    def __init__(self, ax, label, valmin, valmax, valinit=None, valfmt=None, closedmin=True,
+                 closedmax=True, dragging=True, valstep=None, orientation="horizontal",
+                 track_color='lightgrey', handle_style=None, **kwargs):
+
+        super().__init__(ax, label, valmin, valmax, valinit=valinit, valfmt=valfmt,
+                         closedmin=closedmin, closedmax=closedmax, dragging=dragging,
+                         valstep=valstep, orientation=orientation, track_color=track_color,
+                         handle_style=handle_style, **kwargs)
+        self.label.set_position((0.5, 0.8))
+        self.label.set_verticalalignment('bottom')
+        self.label.set_horizontalalignment('center')
+        self.label.set_weight('bold')
+        # "Removes" the labels showing the current range
+        self.valtext.set_visible(False)
+
+    def update_range(self, rng, label=None):
```

Comment:
should we add these args into the docstring?

Reply:
Thanks for catching this!  Added.